pkg/config/config.go

```diff
@@ -45,12 +102,374 @@ func concatenateBytes(data ...[]byte) []byte {
 	return bytes
 }
 
-// MarshalOrPanic serializes a protobuf message and panics if this
+// NewCreateChannelTx creates a create channel tx using the provided config.
+func NewCreateChannelTx(profile *Profile, mspConfig *mb.FabricMSPConfig) (*cb.Envelope, error) {
+	var err error
+
+	if profile == nil {
+		return nil, errors.New("profile is empty")
+	}
+
+	channelID := profile.ChannelID
+
+	if channelID == "" {
+		return nil, errors.New("channel ID is empty")
+	}
+
+	// mspconf defaults type to FABRIC which implements an X.509 based provider
+	mspconf := &mb.MSPConfig{
+		Config: protoMarshalOrPanic(mspConfig),
+	}
+
+	ct, err := defaultConfigTemplate(profile, mspconf)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create default config template: %v", err)
+	}
+
+	newChannelConfigUpdate, err := newChannelCreateConfigUpdate(channelID, profile, ct, mspconf)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create channel create config update: %v", err)
+	}
+
+	newConfigUpdateEnv := &cb.ConfigUpdateEnvelope{
+		ConfigUpdate: protoMarshalOrPanic(newChannelConfigUpdate),
+	}
+
+	env, err := createEnvelope(cb.HeaderType_CONFIG_UPDATE, channelID, newConfigUpdateEnv)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create envelope: %v", err)
+	}
+
+	return env, nil
+}
+
+// protoMarshalOrPanic serializes a protobuf message and panics if this
 // operation fails.
-func MarshalOrPanic(pb proto.Message) []byte {
+func protoMarshalOrPanic(pb proto.Message) []byte {
 	data, err := proto.Marshal(pb)
```

Comment:
we might not want to panic at all in this package?
https://github.com/hyperledger/fabric/pull/669#discussion_r379832988

Reply:
addressed in latest commit